app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -68,11 +68,30 @@
 
     private boolean batterySaver;
 
-    private boolean pokeFlyRunning = false;
+    /**
+     * This button detects if pokeFly is starting.
+     * Such booleans are error-prone, but we only use it to decide what label to show in launchButton.
+     */
+    private boolean pokeFlyStarting;
+
     private int trainerLevel;
 
+    private Button launchButton;
+
     private final Point arcInit = new Point();
     private int arcRadius;
+    private final BroadcastReceiver pokeflyStateChanged = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (Pokefly.ACTION_START.equals(intent.getAction())) {
+                updateLaunchButtonText(true);
+                pokeFlyStarting = false;
+            } else if (Pokefly.ACTION_STOP.equals(intent.getAction())) {
+                updateLaunchButtonText(false);
+                pokeFlyStarting = false;
+            }
+        }
```

Comment:
1. Combine this into one intent that checks "Pokefly.isRunning()"
2. Let's call that intent "UPDATE_BUTTON"
3. "Enable" the button again (see comment in startPokefly() )